src/components/m-button/index.scss

```diff
@@ -12,6 +12,14 @@
 
 .m-button__icon {
   @include button__icon();
+
+  &:not(.m-button__icon--reverse) {
```

Comment:
personally I'm not a big fan of the `:not` selector, rather I overwrite rules.
But that's just a flavor of mine, in fact [caniuse](https://caniuse.com/#search=%3Anot) clearly proves me being paranoid and over skeptical...

Reply:
Though we should check how much `:not` affects CSS specificity if our goal is zero specificity CSS #383